src/run/run.c

```diff
@@ -2647,6 +2669,36 @@ static int start_transient_service(sd_bus *bus) {
                 if (r < 0)
                         return r;
 
+                sd_event_source **forward_signal_sources = NULL;
+                size_t n_forward_signal_sources = 0;
+                CLEANUP_ARRAY(forward_signal_sources, n_forward_signal_sources, event_source_unref_many);
+
+                /* There's no guarantee that we're permitted to forward signals to the service main process,
+                 * but there's no harm in trying. */
+                r = event_forward_signals(
+                                c.event,
+                                &c.pidref,
+                                pty_forward_signals, ELEMENTSOF(pty_forward_signals),
+                                &forward_signal_sources, &n_forward_signal_sources);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to set up signal forwarding: %m");
+
+                if (c.pty_fd >= 0)
+                        FOREACH_ARRAY(source, forward_signal_sources, n_forward_signal_sources) {
+                                r = sd_event_source_get_signal(*source);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to get event source signal: %m");
+
+                                if (!IN_SET(r, SIGTERM, SIGINT))
+                                        continue;
+
+                                /* This causes the signal forwarding event source to terminate the event loop
+                                 * with a zero exit code if a signal can't be forwarded. */
+                                r = sd_event_source_set_exit_on_failure(*source, true);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to enable exit-on-failure: %m");
+                        }
```

Comment:
so, hmm, this is also set up when using --wait? I'd really prefer it's only done for interactive uses. I mean, I very often spawn stuff with --wait and interrupt the client... Plus, I'm pretty sure if the user specifies --wait --pty and disconnects forwarder with hotkeys we should disable forwarding. As commented in pty_forward_handler we want SIGINT to interrupt the (waiting) client, and the propagation would be quite a compat break and surprising.